src/manager.rs

```diff
@@ -0,0 +1,194 @@
+    pub fn set_enabled(&mut self, name: &String, enabled: bool) -> Result<(), Error> {
+        let source = try!(self.get_source(name));
+        let is_running = self.runners.contains_key(name);
+        if !enabled && is_running {
+            if let Some(mut runner) = self.runners.remove(name) {
+                let (tx, rx) = channel();
+                runner.stop(move |result| {
+                    tx.send(result).unwrap();
+                });
+                let _ = try!(rx.recv());
+            }
+        } else if enabled && !is_running {
+            try!(self.start_script(name, &source));
+        } else {
+            // Already running, or already not running.
+        }
+
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        try!(connection.execute("UPDATE scripts SET is_enabled = $1 WHERE name = $2",
+                                &[&enabled, name]));
+        Ok(())
+    }
+
+    /// Remove a script entirely, stopping it if necessary.
+    pub fn remove(&mut self, name: &String) -> Result<(), Error> {
+        try!(self.set_enabled(name, false));
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        connection.execute("DELETE FROM scripts WHERE name = $1", &[name])
+            .map(|_| ())
+            .map_err(Error::RusqliteError)
+    }
+
+    // Remove all scripts, stopping any running scripts.
+    pub fn remove_all(&mut self) -> Result<(), Error> {
+        // Get a copy of the keys (so that we don't borow `self` twice).
+        let keys: Vec<String> = self.runners.keys().map(|k| k.clone()).collect();
+        // Remove any running scripts.
+        for name in keys {
+            try!(self.remove(&name));
+        }
+        // Nuke the scripts database.
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        connection.execute("DELETE FROM scripts", &[])
+            .map(|_| ())
+            .map_err(Error::RusqliteError)
+    }
+
+    /// Get the number of currently-running scripts.
+    pub fn get_running_count(&self) -> usize {
+        self.runners.len()
+    }
+
+    /// Get the source for a script with the given name.
+    fn get_source(&self, name: &String) -> Result<String, Error> {
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        let mut stmt = try!(connection.prepare("SELECT source FROM scripts WHERE name = $1"));
+        let mut rows = try!(stmt.query(&[name]));
+        let first_row = try!(try!(rows.nth(0).ok_or(Error::NoSuchScriptError)));
+        Ok(first_row.get(0))
+    }
+
+    /// Execute a script. Returns an error if the script is already running.
+    fn start_script(&mut self, name: &String, source: &String) -> Result<(), Error> {
+        if self.runners.contains_key(name) {
+            return Err(Error::ScriptAlreadyRunning);
+        }
+        let mut runner = Execution::<Env>::new();
+        let (tx, rx) = channel();
```

Comment:
Instead of defining this channel yourself, let's move the problem to the higher level and:
- add as argument to `fn new<T>(...)` a `tx: T` where `T: Box<ExtSender<(Id<ScriptId>, ExecutionEvent)>>`;
- in `start_script`, use `map` to convert the `ExtSender(ExecutionEvent)` expected by `runner.start` into a `ExtSender<(Id<ScriptId>, ExecutionEvent)>`;
- document `new` to make sure that the contents of the `tx` are consumed, otherwise there will be a memory leak.

(see also comment below for the result of `runner.start`)